### PR TITLE
Add removeprefix/removesuffix to Starlark strings

### PR DESCRIPTION
**Pull request by fmeum**
Implements https://github.com/bazelbuild/starlark/issues/185 in Java Starlark.

### CODE REVIEW

**Comment by fmeum** [comment]
This is ready for review now that https://github.com/bazelbuild/starlark/commit/d6a52c5fd1bd0e7010fb9dbf426b54a2d462af66 has landed.

**Review by adonovan — APPROVED** [review]
LGTM, though you'll need @brandjon to merge it upstream.

File: src/test/java/net/starlark/java/eval/testdata/string_misc.star [hunk 1]
```diff
@@ -177,3 +177,23 @@ assert_eq("abc" * 0, "")
...
+assert_eq("Apricot".removesuffix("cot"), "Apri")
+assert_eq("Apricot".removesuffix("Cot"), "Apricot")
+assert_eq("Apricot".removesuffix("t"), "Aprico")
+assert_eq("a".removesuffix(""), "a")
+assert_eq("".removesuffix(""), "")
+assert_eq("".removesuffix("a"), "")
+assert_eq("Apricot".removesuffix("co"), "Apricot")
+assert_eq("Apricotcot".removesuffix("cot"), "Apricot")
+
+# startswith
```
Comment (adonovan) [inline src/test/java/net/starlark/java/eval/testdata/string_misc.star:191]: `# removeprefix`

Comment (fmeum) [reply src/test/java/net/starlark/java/eval/testdata/string_misc.star]: Thanks, fixed and sorted.

**Comment by fmeum** [comment]
@adonovan I added a failure test and a test for no modifications for each method now that this behavior is also documented in the spec.

**Pull request closed without merging**